lib/cash-in/cash-in-tx.js

```diff
@@ -32,18 +32,28 @@ function post (machineTx, pi) {
   return cashInAtomic.atomic(machineTx, pi)
     .then(r => {
       const updatedTx = r.tx
-      let blacklisted = false
+      let blacklisted = null
       let addressReuse = false
       let walletScore = {}
 
-      return Promise.all([settingsLoader.loadLatest(), checkForBlacklisted(updatedTx), doesTxReuseAddress(updatedTx), getWalletScore(updatedTx, pi)])
-        .then(([{ config }, blacklistItems, isReusedAddress, fetchedWalletScore]) => {
+      return Promise.all([
+        settingsLoader.loadLatest(),
+        checkForBlacklisted(updatedTx),
+        doesTxReuseAddress(updatedTx),
+        getWalletScore(updatedTx, pi)
+      ])
+        .then(([
+          { config },
+          blacklistItems,
+          isReusedAddress,
+          fetchedWalletScore
+        ]) => {
           const rejectAddressReuse = configManager.getCompliance(config).rejectAddressReuse
 
           walletScore = fetchedWalletScore
 
           if (_.some(it => it.address === updatedTx.toAddress)(blacklistItems)) {
-            blacklisted = true
+            blacklisted = _.find(it => it.address === updatedTx.toAddress)(blacklistItems)
```

Comment:
This can be improved to a single `_.find()`:

```js
blacklisted = _.find(...)
if (blacklisted) {
  notifier.notifyIfActive(...)
}
```

Alternatively, if `blacklist.blocked()` is changed to a `oneOrNone()`, the `_.find()` can be dropped.